### DISCUSSION TITLE
How can I know specfication of non-standard functions of z3

### DISCUSSION THREAD

**Discussion by hagozaebii** [Z3Prover/z3#7290] [Q&A]
Hello, I am a Z3 user.

To my knowledge, Z3 provides non-standard functions (e.g., seq.last_indexof) as well. Could you please advise me on where I can find a list of these non-standard functions and their usage instructions?

**Comment by NikolajBjorner** [comment]
this one is not documented.
The functions are listed here:
https://github.com/Z3Prover/z3/blob/9073da4ee63abd4ea069a4444db6d1d66da57ed0/src/ast/seq_decl_plugin.cpp#L214

The documented functions are listed here:

https://microsoft.github.io/z3guide/docs/theories/Strings

and here

https://microsoft.github.io/z3guide/docs/theories/Sequences

**Reply by hagozaebii** [reply]
Thanks for your answer @NikolajBjorner 
I think that there are non-standard functions which are not involved with seq.
Could I know that How to find them?
